buildSrc/src/main/kotlin/HashUtils.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// This code has been adapted from https://gist.github.com/LongClipeus/84db46e7d9714f67c4cbc40a67c8be1e
```

Comment:
What's the license on this?

Reply:
I don't know. [I've asked the author](https://gist.github.com/LongClipeus/84db46e7d9714f67c4cbc40a67c8be1e?permalink_comment_id=4185034#gistcomment-4185034).

Reply:
Turns out we can accomplish the same thing as this code provided the file is under 2GB in one line of Kotlin, see https://github.com/awslabs/smithy-rs/pull/1422/commits/ef450f12b3373565a8dcb2c58fd2e261b1f5f29a.